StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/LayoutRules/SA1516CSharp9UnitTests.cs

```diff
@@ -55,5 +55,64 @@ public async Task TestStatementSpacingInTopLevelProgramAsync()
                 FixedCode = fixedCode,
             }.RunAsync(CancellationToken.None).ConfigureAwait(false);
         }
+
+        /// <summary>
+        /// Verifies that SA1516 is not reported between global statement in top-level programs.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        [WorkItem(3351, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3351")]
+        public async Task TestGlobalStatementSpacingInTopLevelProgramAsync()
+        {
+            var testCode = @"int i = 0;
+return i;
+";
+
+            await new CSharpTest(LanguageVersion.CSharp9)
+            {
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                TestState =
+                {
+                    OutputKind = OutputKind.ConsoleApplication,
+                    Sources = { testCode },
+                },
+            }.RunAsync(CancellationToken.None).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that SA1516 is reported between global statement and record declaration in top-level programs.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestGlobalStatementAndRecordSpacingInTopLevelProgramAsync()
+        {
+            var testCode = @"return 0;
+{|#0:record|} A();
+";
+
+            var fixedCode = @"return 0;
+
+record A();
+";
+
+            await new CSharpTest(LanguageVersion.CSharp9)
+            {
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                TestState =
+                {
+                    OutputKind = OutputKind.ConsoleApplication,
+                    Sources = { testCode },
+                    ExpectedDiagnostics =
+                    {
+                        // /0/Test0.cs(2,1): warning SA1516: Elements should be separated by blank line
+                        Diagnostic().WithLocation(0),
+
+                        // /0/Test0.cs(2,1): warning SA1516: Elements should be separated by blank line
+                        Diagnostic().WithLocation(0),
```

Comment:
❔ Do we know why this is reported twice?

Reply:
It's also reported twice in the first test in this file (from which I copied this one), so it's an existing behavior.
Not sure why it's reposted twice (or even is it actually reported twice, or it's a test system issue)